### DISCUSSION TITLE
Ports suggestions

### DISCUSSION THREAD

**Discussion by symbios24** [Exarkuniv/RetroPie-Extra#3] [Ideas]
Hi there

For first suggestion i have OpenRA

https://github-wiki-see.page/m/OpenRA/OpenRA/wiki/OpenRA-on-RaspberryPi

Q3lite
quake3 game engine capable of playing urban terror and smoking guns

https://github.com/cdev-tux/q3lite/wiki/Compiling-Install-Guide

boswars port

install with apt get install boswars and run normal with boswars

Berusky port

apt install berusky xorg

Put in the berusky executable script: startx berusky

More will come soon

**Comment by Exarkuniv** [comment]
@symbios24 thanks for the ideas, 
i already have one for OpenRA. so you good there. the frame rate is a little lacking but it does work.

ill add the others to my TODO :)

**Reply by symbios24** [reply]
I will give you more probably tomorrow like wesnoth and some others,i see a message in your github that says exclude your issues with author symbios24
Whats that?

**Comment by Exarkuniv** [comment]
@symbios24 i have no idea. can you take a screen shot so i can see where it says that. it might help me find out whats going on. 
still new with all that github can do

**Reply by symbios24** [reply]
Its under the ussues section it says Protip

**Reply by Exarkuniv** [reply]
its just a tip. if you wanted to exclude things.
it changes everytime. i even got one to exclude myself

**Comment by actraiser30** [comment]
Wesnoth

install: apt install wesnoth,xorg,timidity,freepats

emulators.cfg file in \opt\retropie\configs\ports\wesnoth

wesnoth = "xinit /usr/games/wesnoth"
default = "wesnoth"

executable script in ports:

#!/bin/bash
"/opt/retropie/supplementary/runcommand/runcommand.sh" 0 _PORT_ "wesnoth" ""

**Comment by actraiser30** [comment]
some scripts more ready to put in scriptmodules ports

https://easyupload.io/t305yd

**Comment by actraiser30** [comment]
simutrans

apt install simutrans timidity freepats xorg

emulators.cfg file in \opt\retropie\configs\ports\simutrans

simutrans = "xinit /usr/games/simutrans -fullscreen -res 4"  
default = "simutrans"

executable script in ports:

#!/bin/bash
"/opt/retropie/supplementary/runcommand/runcommand.sh" 0 _PORT_ "simutrans" ""

**Comment by actraiser30** [comment]
i would suggest to put this ports also by removing the game files if any.

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/openclaw.sh

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/sqrxz4.sh

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/avp.sh

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/vvvvvv.sh

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/supaplex.sh

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/fruity.sh

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/temptations.sh

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/spelunky.sh

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/openrct2.sh

https://github.com/jmcerrejon/PiKISS/blob/master/scripts/games/hermes.sh

plex media server

https://github.com/bdevans/plexberry/blob/master/setup_plex.sh

https://gist.github.com/NuroDev/f9354dd91190b58daae7b05cf3ea56ce

torrent server

https://github.com/transmission/transmission

Minecraft server script

http://www.nickmasso.com/projects/mc-install

**Reply by Exarkuniv** [reply]
@actraiser30 thanks for all of those. i will take a look at them. 
as for the PiKiss ones. if they are installing copyright stuff. im not going to use them. I was already hit with a DMCA once and had something taken down. i would not like for a repeat, but i will gladly look

**Reply by symbios24** [reply]
I think only the avp and openclaw have only copyrighted stuff but off course you can still put this two ports by excluding/remove the copyrighted files

**Comment by Exarkuniv** [comment]
@i have to look at open claw. unless it has changed you need the copyrighted stuff to finish compiling the binany. but when i got it working it was over a year ago. 
as i said i will take a look at them all

**Reply by symbios24** [reply]
If you need the avp compiling procedure,i have created a script that does this only

**Reply by Exarkuniv** [reply]
if you want to give me it, i would be grateful. if not, im sure ill figure it out. in time

**Reply by actraiser30** [reply]
https://easyupload.io/st9350

after build:

emulators.cfg file in \opt\retropie\configs\ports\avp

inside:
avp = "xinit /home/pi/avp/./avp -f"  (put the location of the compiled executable, this one has the location i that i had build it)
default = "avp"

executable script in ports:

#!/bin/bash
"/opt/retropie/supplementary/runcommand/runcommand.sh" 0 _PORT_ "avp" ""

**Reply by symbios24** [reply]
Sorry i have two github accounts one on the phone and one in the laptop

**Reply by Exarkuniv** [reply]
@symbios24 well that makes more sense. and no worries

**Reply by symbios24** [reply]
![IMG_20210917_214258](https://user-images.githubusercontent.com/10866601/149382583-7643b218-450c-4c5d-af35-02886cd813f5.jpg)

One of this days i will try to give you AC

**Reply by Exarkuniv** [reply]
very cool

**Comment by symbios24** [comment]
ii will give later a fix for nxengine if you did not manage yet to make it work

**Reply by Exarkuniv** [reply]
lol. im working on that right now. but im not bothering with the old version. im working on getting a newer engine to work. 
but if you want to take a crack at the old one go for it. i have no issues having different ports for things

**Reply by actraiser30** [reply]
not really a crack its another version maybe is the same as the one you working now

https://github.com/PokeEngineer/NXEngine-RPi   is that the one you do now?

**Reply by Exarkuniv** [reply]
thats a dead link so, we have to make our own

**Reply by actraiser30** [reply]
i dont think so

**Reply by actraiser30** [reply]
i predicted everything

https://easyupload.io/77g6hn

**Reply by Exarkuniv** [reply]
if you want to make something from that, go for it. im working on building from source for NXEngine-evo. 

and my point still stands. we have to make our own. meaning we have to host it and make a script.

**Reply by symbios24** [reply]
A ok then i did not know that

**Reply by actraiser30** [reply]
i have also the source code in a zip if you want so you can use it for host and build

**Reply by actraiser30** [reply]
anyway if an old port is not working let me know maybe i have it already fix it

**Reply by Exarkuniv** [reply]
no the old nxengine port does not work. it fails when it starts to compile

**Reply by actraiser30** [reply]
yes i know you can send a message to the developer

https://nxengine.sourceforge.io/

interface-nx@starlightstarbright.org

**Reply by Exarkuniv** [reply]
i know, but since there has not been a update on the page since 2016, im going with hes not doing anything with it anymore.

**Comment by actraiser30** [comment]
Exar how old are you?
im 44
any real consoles?
i have an snes,megadrive and nes

**Reply by Exarkuniv** [reply]
I'm 40
I only have Sega master system for retro systems
But I have all the modern ones.

I'm working on getting some more retro ones

**Reply by actraiser30** [reply]
You do that they worth it,nothing is compared with the real thing,i think the snes or megadrive classics are between retro and modern ones,they worth taking.they play many systems like they suppose to be.

**Reply by symbios24** [reply]
![super-game-boy](https://user-images.githubusercontent.com/10866601/149578299-1e6c6b85-b14f-491b-bd8c-1c4d8ff1050a.jpg)

**Comment by symbios24** [comment]
Is openxcom working now?

**Reply by Exarkuniv** [reply]
have not gotten to that one. 
im focused on getting nxengine-evo 2.6.5 working

**Reply by symbios24** [reply]
If its not working leave it,they must be other nx engines out there.

**Reply by Exarkuniv** [reply]
one thing about me. i never leave it till i find out that it cant run at all on my Pi. 
if i just gave up when i could not figure something out about 90% of my scripts that i made would not be there. 

also i do my scripts while im at work. so there are times i dont get much done in a day. what gets done gets done.

**Reply by symbios24** [reply]
They need time,i had many abandoned things but also restart things,i abandon AC and n 2017 and in 2021 in 5 mins i finish it
Practically i never leave things just pause them for a while
Whats your work, programmer?
Im computer technician.
Did you got the zero2?
Shame about the ram size

**Reply by symbios24** [reply]
I go to sleep maybe i can give a try tomorrow to the cave story
What the git?

**Reply by Exarkuniv** [reply]
if you want to get the old version working go for it, i just got the one i was working on done and working.
already updated it on github

im a electronics technician

**Reply by symbios24** [reply]
I have an electronics diploma never practice it
Well done with nx,you did it in raspbian buster or bullseye?

**Reply by Exarkuniv** [reply]
Everything is in buster. I make all my scripts on a current RetroPie build.

**Reply by symbios24** [reply]
Hi there,any change that you can add this to the scripts?

Install instructions

https://github.com/henrikstengaard/hstwb-installer/wiki/Install-HstWB-Installer-for-Raspberry-Pi-OS-Lite

Git

https://github.com/henrikstengaard/hstwb-installer

Thanks

**Reply by Exarkuniv** [reply]
i will give it a crack

**Comment by symbios24** [comment]
Some more suggestions

friking-shark

https://github.com/ptitSeb/friking-shark

Call to power2

https://github.com/ptitSeb/ctp2

stuntcarremake

https://github.com/ptitSeb/stuntcarremake

Treadmarks

https://github.com/ptitSeb/TreadMarks

Kamremake

https://github.com/ptitSeb/kam_remake

Updated Hydra Castle Labyrinth port

https://github.com/s1eve-mcdichae1/HCL-RPi

**Comment by symbios24** [comment]
Hello there
Can you add this script that creates a minecraft bedrock server?

https://github.com/TheRemote/MinecraftBedrockServer

**Comment by actraiser30** [comment]
Hello there

I saw that you removed some good ports like nblood,dunelegacy etc.

why is that?

**Comment by Exarkuniv** [comment]
they will be back, im reworking them and it seems everytime i start working on one i end up getting help. and then end up not doing anything since the peoiple helping me do it all. 

i would like to learn the better way of writing scripts but for me. having it done does not help me. so i picked the one that i made myself for me to rework and then ill add them again

**Reply by actraiser30** [reply]
ok if you need help in anything that i might know tell me

thanks

**Comment by symbios24** [comment]
I recently got an Odroid C4 and installed Retropie following the C2 guide,needs more testing though
Do you have any script that simplify the retropie installation in odroid boards?
Or is there a ready to use image for C4 with Retropie?

**Reply by Exarkuniv** [reply]
i dont know of any right now. the one group that might have on in the future would be https://techtoytinker.com/ since he has a few different boards. and i have used the image for the XU4 and Jetson NANO from him and they do work. 
other then that, i dont know of any

**Reply by symbios24** [reply]
Thanks it looks like not many images out there for c4,xu4 and n2+ are very popular
I will be experimenting with Retropie in c4
I tried batocera but was not like retropie that you can edit things.since im using only command line and not desktop, do you know how to enable the gpu in cli(command line)? I Installed a bunch of games and they all ask for the gpu,driver,opengl etc,in Raspberry pi this was never a problem most of them played right away from cli.
Thank you

**Comment by Exarkuniv** [comment]
sadly i do not

**Comment by actraiser30** [comment]
hi there is it possible to add the fuse (zx spectrum) standalone emulator in your ports?

as the Retropie version of the emulator is outdated 1.6.0 is the newest 1.5.7 Retropie has

https://github.com/RetroPie/RetroPie-Setup/blob/master/scriptmodules/emulators/fuse.sh

i think you need only to change the 2 links in the script from fuse-1.5.7.tar.gz to fuse-1.6.0.tar.gz

and add it in your scripts for example as fuse-1.6.0 so it does not conflict with the older version

thank you

**Reply by Exarkuniv** [reply]
yes, that can be done easily

**Reply by Exarkuniv** [reply]
@actraiser30 ok i added with the 2 changes, if you can check and make see if its working

**Comment by actraiser30** [comment]
thanks a lot i will try 

i forgot to give you another two only when you have time

one is vice (c64) that its also outdated

https://github.com/RetroPie/RetroPie-Setup/blob/master/scriptmodules/emulators/vice.sh

you just need to replace in the line " rp_module_repo="svn svn://svn.code.sf.net/p/vice-emu/code/tags/v3.5/vice - HEAD"

to rp_module_repo="svn svn://svn.code.sf.net/p/vice-emu/code/tags/v3.6.1/vice - HEAD"

and the other is cpcemu (amstrad)

https://cpc-emu.org/Release/2021-12-29/cpcemu-linux-arm-2.4.tar.gz

that is require sdl2

thank you

**Reply by actraiser30** [reply]
i miss an archive in fuse.sh in the line "downloadAndExtract "$__archive_url/libspectrum-1.4.4.tar.gz" "$md_build/libspectrum" --strip-components 1"

it should be :downloadAndExtract "$__archive_url/libspectrum-1.5.0.tar.gz" "$md_build/libspectrum" --strip-components 1

thanks

**Reply by Exarkuniv** [reply]
ok i updated fuse

**Reply by Exarkuniv** [reply]
i also added vice with your fix

**Reply by actraiser30** [reply]
have you put vice and fuse in optional packages or in experimental?

i cannot seem to find them in experimental and the vice from optional installs the old version do i need to do anything else?

**Reply by Exarkuniv** [reply]
@actraiser30 they will be in experimental, i had to rename them so the OG scripts from RP did not over write them.
so to update you need to do this 
`cd RetroPie-Extra/`
then 
`./update-extras.sh`

it will install the newest version, but you will have to manually remove the old ones,

they sit in `/home/pi/RetroPie-Setup/ext/RetroPie-Extras/scriptmodules/emulators` and delete the `fuse.sh` and `vice.sh`

**Comment by actraiser30** [comment]
i manage to find then they were in optional packages but they were mixed-up their menu it had first vice then an empty space and then a 1.6 only,it looks like they conflicting with the ones retropie has,can you put them in the experimental in the section Retropie Extras Emulators they probably need a different name to work

when i click the 1.6 it does nothing, the fuse failed to install it cannot find the packages i gave you
the vice is installed but its still the old version,anyway better to remove them they are not working,thanks for trying

**Reply by Exarkuniv** [reply]
i missed the part where the script would be, that is now fixed, as the rest of it. i dont know. 

i changed what you said, and how you said it

**Reply by Exarkuniv** [reply]
i am working on it, got some of it cleared up, but not all

**Reply by Exarkuniv** [reply]
so it looks like `fuse` will be a bit more work, as the files we are looking for are not hosted by RP, like the other ones so they cant be downloaded that way

im working on finding them for me to link to download and install. so `fuse` will be a bit more work.

`Vice` so far is installing fine, i will update once it is done

**Reply by actraiser30** [reply]
thanks they are in experimental now
fuse its not working cannot find the changed links
and vice is working but its still the old version something must be wrong with the changed url svn.code.sf.net/p/vice-emu/code/tags/v3.6.1/vice

probably this command takes the 3.5 instead the 3.6 :     svn checkout "$md_repo_url" "$md_build"

**Comment by Exarkuniv** [comment]
`fuse its not working cannot find the changed links`
thats what i said, lol

**Reply by symbios24** [reply]
Sorry i was dizzy im not that young anymore

**Reply by Exarkuniv** [reply]
it all good, lol

**Comment by symbios24** [comment]
I wonder how many decades we will wait to play this on the raspi

https://m.youtube.com/watch?v=LZ8fZ4cKa1U

😂

**Reply by Exarkuniv** [reply]
i dont think we will be around then, and if we are, we will not be able to afford it at this rate 
lol

**Comment by actraiser30** [comment]
Hello when you will put back this removed ports?

vanillacc.sh
score.sh
rednukem.sh
pcexhumed.sh
nblood.sh
hhexen.sh
hheretic.sh
dunelegacy.sh
dunedynasty.sh
cytadela.sh
augustus.sh

**Reply by Exarkuniv** [reply]
you know you can go and just download them all from their repos that i have set up 
exp
on this page https://github.com/Exarkuniv/Augustus-RPI
you just run 
``wget https://raw.githubusercontent.com/Exarkuniv/Augustus-RPI/Master/augustus.sh -P $HOME/RetroPie-Setup/ext/scriptmodules/ports/``

and it will install it

**Reply by symbios24** [reply]
I know that but its better to have them all in one place

**Comment by actraiser30** [comment]
Hello

rocksndiamonds has been updated can you change on the script the " rocksndiamonds-4.2.3.1.tar.gz "
to rocksndiamonds-4.3.2.1.tar.gz

also a question in the hexen2 you have on the script the gl version or the normal one?

i saw it now by opening the script its the gl version can you add also the non-gl version i tried by renaming the script but whatever name i gave it when i open the retropie-script i got a message that it conflicts with the other one.

thanks

**Reply by symbios24** [reply]
Sorry i give you too much sometimes only when you have time take a look at this if you can do scripts for them

Q3lite

https://github.com/cdev-tux/q3lite/wiki/Compiling-Install-Guide

Hstwb 

https://github.com/henrikstengaard/hstwb-installer/wiki/Install-HstWB-Installer-for-Raspberry-Pi-OS-Lite

**Reply by Exarkuniv** [reply]
When I'm back from vacation I will take a look

As for hexen2 I did not write the script to start with. 

But I can take a look and if it's easy I'll make a new script

**Comment by actraiser30** [comment]
Enjoy your vacations 😉

Στις Κυρ, 5 Ιουν 2022, 20:31 ο χρήστης Exarkuniv ***@***.***>
έγραψε:

**Comment by Exarkuniv** [comment]
rocksndiamonds has been updated can you change on the script the " rocksndiamonds-4.2.3.1.tar.gz "
to rocksndiamonds-4.3.2.1.tar.gz

switching it causeing the script to fail, i have reached out to the Dev and see what they say about it, till then it will stay as is

**Reply by Exarkuniv** [reply]
got a answer back, we need a newer version of SDL2 2.0.14 or higher is needed, 
will need to look in to installing a version manually

**Reply by symbios24** [reply]
Or this or just try one of the other newer but not latest version to see which one works without sdl 2.0.14
At least to have it somehow updated.

Or put a warning that sdl 2.0.14 or newer needs to be installed first from the retropie-setup dependencies menu for the game to work

**Reply by Exarkuniv** [reply]
unless you know how to install newer SDL it will not work on buster. maybe once OS64 is supported by Retropie.

when i have time i might see if a newer version will work, for now i will open a issue (which you can do also) so if it not forgotten

**Comment by symbios24** [comment]
Hello there

Can you put in your script the standalone ps1 emulator Duckstation?

https://github.com/stenzek/duckstation

Compiling instructions:

https://github.com/stenzek/duckstation#building

It runs perfect with matchbox and xinit

Thanks

**Reply by Exarkuniv** [reply]
i already had a script for libretro duckstaion for awhile now, that one does not use matchbox and xinit

i have used it lightly and seems to work fine for what i have seen, not sure is matchbox and xinit is really needed. 

if there is a big improvement to it, i will try to take a look at it

**Reply by symbios24** [reply]
This is a standalone emulator not libretro,i prefer that king of emulators most since they dont use a pre environment to work,is it possible,to make and give me here the contents of a script for this non libretro emulator,so i can put it in my scripts? As i said it uses xinit(xorg) matchbox like many ports do
Thanks

**Reply by Exarkuniv** [reply]
when i have time to give this a try, i will,

**Reply by symbios24** [reply]
Hello

Can you also add the portal expansion of hexen2 to the normal port of hexen2 not the gl one as i use that,and if possible add also the normal non gl hexen2 to your extras scripts
Thank you

**Comment by symbios24** [comment]
Hello

Can you also add the portal expansion of hexen2 to the normal port of hexen2 not the gl one as i use that,and if possible add also the normal non gl hexen2 to your extras scripts
Thank you

**Reply by Exarkuniv** [reply]
that script was updated with that earlier this morning with that. 
it is to just the only hexen2 script i have at the moment.

**Comment by symbios24** [comment]
Is it possible to add a script tha creates a custom berryboot installer??

I mean a script that creates an updated berryboot version and add it to extras 

I have the instructions if you want

**Reply by Exarkuniv** [reply]
not sure how that will work since berryboot is a OS installer, and to run the scripts you need a OS already running. 

if you want you can post the instructions and i can look at them, like i said not sure how it will work, if it will

**Reply by symbios24** [reply]
The script will create the berryboot installer that your can copy later to an empty memoty card, it wont install berryboot anywhere

**Reply by Exarkuniv** [reply]
does that even work on a Pi?

**Reply by symbios24** [reply]
We can just try it should work
I saw that you enchange the script,gui etc
can you have the option for the old simple installer also,eg: the normal one time install placing the scriptmodules in etc..

Thank you

**Reply by Exarkuniv** [reply]
maybe, i can see how easy it is to do something with out making it too messy, 
having 2 different ways to do the same thing, is just silly for me, so it might not happen. 

Im looking to see if we can add a way to do a mass install for each group. 

 im not the one who wrote it, im trying to keep things clean and simple. and picking what to install is just that
 we went that way since this now have a lot more scripts then what zerojay had so it takes a little bit to get into the RP menu, and i will be adding more
 
i like the idea with picking what to install for the fact most people are going to only want the one thing like the ports, and thats it. so why bother with the libretrocores, or supplementary

**Reply by symbios24** [reply]
I understand that it evolves,in my opinion it was better the simple way,simple and fast 

How can i clone the last commit that has the last version without the gui ??
Thanka

**Reply by actraiser30** [reply]
i want to install extras from the commit: eb339788 i know its possible but i dont know the commands to do so

**Comment by Exarkuniv** [comment]
i have added the old install script to the main git

```
cd ~
git clone https://github.com/Exarkuniv/RetroPie-Extra.git
cd RetroPie-Extra/
./mass-install-extras.sh
```

everything will be as before

**Reply by Exarkuniv** [reply]
i was trying to find a way to install the older commit also, but i could not get the commands right.

this way ended up way easier

**Reply by symbios24** [reply]
Thanks a lot,until what commit it installs?

**Reply by Exarkuniv** [reply]
that is just the old install script. you run it as you have every time before

but right now im having a issue with it. the Permission is messed up. so hold on

**Reply by symbios24** [reply]
Okies hope you can solve it

**Reply by Exarkuniv** [reply]
ok i got it now.

```
cd ~
git clone https://github.com/Exarkuniv/RetroPie-Extra.git
cd RetroPie-Extra/
./mass-install-extras.sh
```

**Reply by symbios24** [reply]
Great you are genius 😸

**Comment by symbios24** [comment]
Good morning

I might be asking a lot,is it possible that you take the: https://github.com/maxnet/berryboot
Repository the creator does not seem to be active for a year in that one,he is active though on other projects,there are also over 300+ issues unanswered, berryboot has a lot of potential and is very popular over the years, it just needs the right person to improve it etc.i dont think the people that have it to bother anymore.

Thanks in advance

**Reply by Exarkuniv** [reply]
Yeah, as much as i would love to say i can do that, that is way above my head there.  Sorry

**Reply by symbios24** [reply]
You are doing great anyway thanks

**Comment by actraiser30** [comment]
Eather way if i had the script i will try it right away with my heavy
overclocked Rpi4 with hdd boot

Στις Παρ, 15 Ιουλ 2022, 20:54 ο χρήστης G D ***@***.***> έγραψε:

**Comment by actraiser30** [comment]
If its a linux it works,Most probably will take forever, :p or not

Στις Παρ, 15 Ιουλ 2022, 19:48 ο χρήστης Exarkuniv ***@***.***>
έγραψε:
